plutus-ledger-api/src/Plutus/V1/Ledger/EvaluationContext.hs

```diff
@@ -8,39 +10,194 @@ module Plutus.V1.Ledger.EvaluationContext
     , toMachineParameters
     , costModelParamNames
     , CostModelApplyError (..)
+    , ParamName -- abstract
+    , enumParamNames
     ) where
 
 import Plutus.ApiCommon
-import PlutusCore.Evaluation.Machine.BuiltinCostModel as Plutus
 import PlutusCore.Evaluation.Machine.CostModelInterface as Plutus
-import PlutusCore.Evaluation.Machine.ExBudgetingDefaults as Plutus
-import PlutusCore.Evaluation.Machine.MachineParameters as Plutus
 
-import Barbies
-import Control.Lens
-import Data.Map as Map
-import Data.Maybe
-import Data.Set as Set
+import Control.Monad.Except
+import Data.Ix
 import Data.Text qualified as Text
+import GHC.Generics
 
 -- | The set of valid names that a cost model parameter can take for this language version.
--- It is used for the deserialization of `CostModelParams`.
-costModelParamNames :: Set.Set Text.Text
-costModelParamNames = Map.keysSet $ fromJust $ extractCostModelParams $
-   defaultCekCostModel
-   & builtinCostModel
-   -- here we rely on 'Deriving.Aeson.OmitNothingFields'
-   -- to skip jsonifying any fields which are cleared.
-   %~ omitV2Builtins
-  where
-    -- "clears" some fields of builtincostmodel by setting them to Nothing. See 'MCostingFun'.
-    omitV2Builtins :: BuiltinCostModel -> BuiltinCostModelBase MCostingFun
-    omitV2Builtins bcm =
-            -- transform all costing-functions to (Just costingFun)
-            (bmap (MCostingFun . Just) bcm)
-            {
-              -- 'SerialiseData','EcdsaSecp256k1',SchnorrSecp256k1 builtins not available in V1
-              paramSerialiseData = mempty
-            , paramVerifyEcdsaSecp256k1Signature = mempty
-            , paramVerifySchnorrSecp256k1Signature = mempty
-            }
+-- It is used for the deserialization of `CostModelParams` and their order in the list is fixed.
```

Comment:
Is this true? Do we actually need this? I think we should be able to get rid of it now?

Reply:
Indeed we should be able to get rid of it, but I don't know if the ledger team uses it somehow in their tests.

Reply:
Part of the point is to force them not to be sensitive to this, so I think we should get rid of it.